docs/fundamentals/code-analysis/quality-rules/il3003.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "IL3003: 'RequiresAssemblyFilesAttribute' annotations must match across all interface implementations or overrides."
+description: "For all interfaces and overrides the implementation 'RequiresAssemblyFilesAttribute' must match the definition 'RequiresAssemblyFilesAttribute', either all the members contain the attribute or none of them."
+ms.date: 12/20/2021
+ms.topic: reference
+author: jtschuster
+f1_keywords:
+  - "IL3003"
+  - "RequiresAssemblyFiles"
+---
+# IL3003: 'RequiresAssemblyFilesAttribute' annotations must match across all interface implementations or overrides.
+
+|  | Value |
+|-|-|
+| **Rule ID** | IL3003 |
+| **Category** | [SingleFile](singlefile-warnings.md) |
+| **Fix is breaking or non-breaking** | Non-breaking |
+
+## Cause
+
+When publishing as a single file (for example, by setting the PublishSingleFile property in a project to true), interface implementations or derived classes with members that don't have matching annotations of `[RequiresAssemblyFiles]` can lead to calling an unsafe member with the attribute. There are four possible scenarios where the warning can be generated:
```

Comment:
In general, please avoid "Safe"/"Unsafe" wording - it feels too much like security related. For trimming we've been trying to use "trim compatible", we can do the same for single-file.